### PR TITLE
Section verbose name

### PR DESCRIPTION
**Pull request by okolimar**
Added verbose_name of section to admin so it is more readable for users.

Also added filter with verbose name.

### CODE REVIEW

File: dynamic_preferences/admin.py [hunk 1]
```diff
@@ -7,18 +7,57 @@
 from .forms import GlobalSinglePreferenceForm, SinglePerInstancePreferenceForm
 
 
+class SectionFilter(admin.AllValuesFieldListFilter):
+
+    def __init__(self, field, request, params, model, model_admin, field_path):
+        super(SectionFilter, self).__init__(field, request, params, model, model_admin, field_path)
+        parent_model, reverse_path = reverse_field_path(model, field_path)
```
Comment (EliotBerriot) [inline dynamic_preferences/admin.py]: Where is this function coming from?

**Comment by okolimar** [comment]
Oh I am blind sorry I just copied it from my project where I tested it by crating admin that overrides the base one so I forgot to copy this line. Though I would gladly help with the test, I am very much out of the time currently so possibly only next weekend.

If so I checked that in test_app there are no verbose_name for Sections so either we need to add one or add whole preference for this test I don't know which one you would prefer.

**Comment by EliotBerriot** [comment]
There is no rush on my side, so it's perfectly fine if you finish your work on this PR later. You can add a verbose_name in test_app sections, it should not break anything :)

File: tests/test_global_preferences.py [hunk 2]
```diff
@@ -110,6 +110,15 @@ def test_global_preference_view_display_form(self):
         self.assertEqual(
             response.context['registry'], registry)
 
+    def test_global_preference_view_section_verbose_names(self):
+        url = reverse("dynamic_preferences.global")
```
Comment (EliotBerriot) [inline tests/test_global_preferences.py]: You are not actually testing the admin here but a dynamic preference url. I think the patterns for reversing admin urls are more like `admin:dynamic_preferences_globalpreferencemodel_changelist` :)

**Comment by okolimar** [comment]
hopefully last iteration :)

**Comment by EliotBerriot** [comment]
Your test looks fine now, but it's not passing on CI unfortuntely!

You can see it here: https://travis-ci.org/EliotBerriot/django-dynamic-preferences/jobs/295926765

Is it working locally ?

**Comment by EliotBerriot** [comment]
It just came to my mind that maybe you you don't know how to run the tests locally:

```
pip install tox   # this is needed to run the tests
tox -e py35-django-111 # to run the tests with Python 3.5 and django 1.11
tox -e py27-django-111 # to run the tests with Python 3.5 and django 1.11
```

File: dynamic_preferences/preferences.py [hunk 3]
```diff
@@ -38,9 +38,14 @@ def __init__(self, name, verbose_name=None):
...
-            return ''
-        return str(self.verbose_name)
+        if hasattr(self, 'verbose_name'):
+            if not self.verbose_name:
+                return ''
+            return str(self.verbose_name)
+        else:
+            if not self.name:
+                return ''
+            return str(self.name)
```
Comment (EliotBerriot) [inline dynamic_preferences/preferences.py]: As you can see in `__init__`, verbose_name is always present on the object.

You can simplify your code:

```
def __str__(self):
    name = self.verbose_name or self.name
    return str(name) if self.name else ''
```

Comment (EliotBerriot) [reply dynamic_preferences/preferences.py]: This is the code snippet you are referring to:

```
if not self.verbose_name:
    return ''
return str(self.verbose_name)
```

As you can see, it's different from what I suggest: it does not include any fallback to `self.name` :)

Comment (okolimar) [reply dynamic_preferences/preferences.py]: Please do you understand the init code? If the init was ran verbose_name would have to be set so no "fallback" is required it is already taken care of inside __init__. But in test_cannot_instanciate_preference_or_section_with_invalid_name test you are initializing Section with wrong names so verbose_name is not None that your code you included here checks for but it does not exist at all so any access results in exception.

I also think that ckecking for None is not necessary as the Section with None name can't exist I did not want to change that though if some backward compatibility would be broken

Comment (EliotBerriot) [reply dynamic_preferences/preferences.py]: There is actually a Section with `None` as a name, [as you can see here](https://github.com/EliotBerriot/django-dynamic-preferences/blob/develop/dynamic_preferences/preferences.py#L45). It's here for backward-compatibility with preferences declared before sections were introduced.

You're right about the exception part, and the fix in your last commit is fine. I'm really sorry I misread the failure log in the build, it's pretty clear now that you say it :)

I'm merging and releasing today.

So,

**Comment by EliotBerriot** [comment]
Released in https://pypi.python.org/pypi/django-dynamic-preferences/1.4.1

**Pull request merged**